google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -422,12 +424,45 @@ public Bucket update(BucketInfo bucketInfo, BucketTargetOption... options) {
 
   @Override
   public Blob update(BlobInfo blobInfo, BlobTargetOption... options) {
-    return todo();
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+    GrpcCallContext grpcCallContext = GrpcRequestMetadataSupport.create(optionsMap);
+    Object object = codecs.blobInfo().encode(blobInfo);
+    UpdateObjectRequest.Builder updateRequestBuilder =
+        UpdateObjectRequest.newBuilder().setObject(object);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_GENERATION_MATCH),
+        updateRequestBuilder::setIfGenerationMatch);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_GENERATION_NOT_MATCH),
+        updateRequestBuilder::setIfGenerationNotMatch);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_METAGENERATION_MATCH),
+        updateRequestBuilder::setIfMetagenerationMatch);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_METAGENERATION_NOT_MATCH),
+        updateRequestBuilder::setIfMetagenerationNotMatch);
+    ifNonNull(
+        (String) optionsMap.get(StorageRpc.Option.PREDEFINED_ACL),
+        updateRequestBuilder::setPredefinedAcl);
+    updateRequestBuilder.setUpdateMask(
```

Comment:
grabbed this from https://github.com/googleapis/java-storage/pull/1483/files so @frankyn comment is valid here as well. not sure if this will work as expected. We will need to generate these masks for quite a few operations so maybe we can make this a helper method?

Reply:
done.